tbx/core/tests/test_navigation_field.py

```diff
@@ -0,0 +1,27 @@
+from tbx.core.factories import HomePageFactory, StandardPageFactory
+from wagtail.models import Site
+from wagtail.test.utils import WagtailPageTestCase
+
+
+class TestNavigationText(WagtailPageTestCase):
```

Comment:
@ababic would welcome your thoughts on this - I had to pick a model to test with and went for the standard page as we already had a factory set up for that. Is this the right place for the test to live?

Reply:
@helenb I would be inclined to rename this module to something like `test_generic_page_functionality`, so that it can be extended in future to test similar 'things' added via a mixin or base class. 

Standard page is fine for testing, but you don't actually need to save anything to the database here, I don't think. That's only really necessary if you're doing custom things on save or in signal handlers. Where you have something like this - where you just want to know if `nav_text` gives you the right value depending on two field values, you can just 'initialize' a page in memory, as suggested below.

If you can remove the need for database access completely, you can swap out the base class for `django.tests. SimpleTestCase` for an impressive performance boost!

Reply:
Thanks - I have implemented your suggestions and now extending `SimpleTestCase` too.